src/map/Map.js

```diff
@@ -687,18 +687,17 @@ L.Map = L.Evented.extend({
 		};
 
 		if (e.type !== 'keypress') {
-			data.containerPoint = target instanceof L.Marker ?
+			var isMarker = target instanceof L.Marker;
+			data.containerPoint = isMarker ?
 					this.latLngToContainerPoint(target.getLatLng()) : this.mouseEventToContainerPoint(e);
 			data.layerPoint = this.containerPointToLayerPoint(data.containerPoint);
-			data.latlng = this.layerPointToLatLng(data.layerPoint);
+			data.latlng = isMarker ? target.getLatLng() : this.layerPointToLatLng(data.layerPoint);

```

Comment:
@mourner this line is not needed for this patch actually, but `latLngToContainerPoint -> containerPointToLayerPoint -> layerPointToLatLng` does not roundtrip, so I was not able to make a unittest strong enough.
Plus I thought the event latlng in case of a marker should be the exact same as the marker one. Not sure though this is the best way to fix this.